odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1471 @@
+            The linear operator that maps ``x --> <x, grad_f(point)>``.
+        """
+        functional = self
+
+        class DerivativeOperator(Functional):
+            def __init__(self):
+                super().__init__(functional.domain, linear=True)
+
+            self.point = point
+
+            def _call(self, x):
+                return x.inner(functional.gradient(point))
+
+        return DerivativeOperator()
+
+    def translate(self, translation):
+        """Creates a translation of the functional.
+
+        For a given functional ``f`` and an element ``translation`` in the
+        domain of ``f``, this operation creates the functional
+        ``f(. - translation)``.
+
+        Parameters
+        ----------
+        translation : `LinearSpaceVector`
+            Element in the domain of the functional
+
+        Returns
+        -------
+        out : `TranslatedFunctional`
+            The functional ``f(. - translation)``
+        """
+        if translation not in self.domain:
+            raise TypeError('the translation {} is not in the domain of the'
+                            'functional {!r}'.format(translation, self))
+
+        return TranslatedFunctional(self, translation)
+
+    def __mul__(self, other):
+        """Return ``self * other``.
+
+        If ``other`` is an operator, this corresponds to composition with the
+        operator:
+
+            ``func * op <==> (x --> func(op(x))``
+
+        If ``other`` is a scalar, this corresponds to right multiplication of
+        scalars with functionals:
+
+            ``func * scalar <==> (x --> func(scalar * x))``
+
+        If ``other`` is a vector, this corresponds to right multiplication of
+        vectors with functionals:
+
+            ``func * vector <==> (x --> func(vector * x))``
+
+        Note that left and right multiplications are generally different.
+
+        Parameters
+        ----------
+        other : {`Operator`, `LinearSpaceVector`, scalar}
+            `Operator`:
+                The `Operator.range` of ``other`` must match this functionals
+                `Functional.domain`.
+
+            `LinearSpaceVector`:
+                ``other`` must be an element of this functionals's
+                `Functional.domain`.
+
+            scalar:
+                The `Functional.domain` of this functional must be a
+                `LinearSpace` and ``other`` must be an element of the ``field``
+                of this functional's `Functional.domain`. Note that this
+                ``field`` is also this functional's `Functional.range`.
+
+        Returns
+        -------
+        mul : `Functional`
+            Multiplication result
+
+            If ``other`` is an `Operator`, ``mul`` is a
+            `FunctionalComp`.
+
+            If ``other`` is a scalar, ``mul`` is a
+            `FunctionalRightScalarMult`.
+
+            If ``other`` is a vector, ``mul`` is a
+            `FunctionalRightVectorMult`.
+
```

Comment:
Second thought on this, we can remove this. Sphinx creates a reference to the superclass method anyway so this link is redundant.